lib/wordcounter.rb

```diff
@@ -0,0 +1,93 @@
+require 'delegate'
+
+class WordHistogram < SimpleDelegator
+  def initialize(*args)
+    super(Hash.new(0, *args))
+  end
+
+  def to_pretty_s
+    word_freq_pairs = self.sort_histogram
+
+    word_lengths = word_freq_pairs.map { |word, _| word }.map &:length
+    max_length = word_lengths.max
+
+    self.to_table_s(word_freq_pairs, middle_column_width: max_length)
+  end
+
+  protected
+  def sort_histogram
+    self.sort do |pair1, pair2|
+      word1, freq1 = *pair1
+      word2, freq2 = *pair2
+
+      next freq2 <=> freq1 unless freq2 == freq1
+      word1 <=> word2
```

Comment:
All of this pairs and sorting stuff could be a lot simpler.  The histogram is a big hash, right?  Then you can just do `self.sort_by { |_word, count| count }`